### PR TITLE
Bazel 3 add fish completions

### PR DESCRIPTION
**Pull request by Profpatsch**
fish completions and minor refactor (see commit messages)

cc @kalbasit

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/build-managers/bazel/bazel_3/default.nix [hunk 1]
```diff
@@ -507,9 +507,15 @@ stdenv.mkDerivation rec {
     mv ./bazel_src/output/bazel $out/bin/bazel-${version}-${system}-${arch}
 
     # shell completion files
-    mkdir -p $out/share/bash-completion/completions $out/share/zsh/site-functions
-    mv ./bazel_src/output/bazel-complete.bash $out/share/bash-completion/completions/bazel
-    cp ./bazel_src/scripts/zsh_completion/_bazel $out/share/zsh/site-functions/
+    install -Dm644 \
+      ./bazel_src/output/bazel-complete.bash \
+      $out/share/bash-completion/completions/bazel/bazel-complete.bash
+    install -Dm755 \
```
Comment (lukegb) [inline pkgs/development/tools/build-managers/bazel/bazel_3/default.nix:513]: This can be 644 as well.

**Review by lukegb — APPROVED** [review]

**Review by avdv — APPROVED** [review]

File: pkgs/development/tools/build-managers/bazel/bazel_3/default.nix [hunk 2]
```diff
@@ -507,9 +507,15 @@ stdenv.mkDerivation rec {
...
-    cp ./bazel_src/scripts/zsh_completion/_bazel $out/share/zsh/site-functions/
+    install -Dm644 \
+      ./bazel_src/output/bazel-complete.bash \
+      $out/share/bash-completion/completions/bazel/bazel-complete.bash
+    install -Dm644 \
+      ./bazel_src/scripts/zsh_completion/_bazel \
+      $out/share/zsh/site-functions/_bazel
+    install -Dm644 \
+      ./bazel_src/scripts/fish/completions/bazel.fish \
+      $out/share/fish/vendor_completions.d/bazel.fish
```
Comment (danieldk) [inline pkgs/development/tools/build-managers/bazel/bazel_3/default.nix:518]:
```suggestion
    installShellCompletion bazel_src/output/bazel-complete.bash
    installShellCompletion bazel_src/scripts/fish/completions/bazel.fish
    installShellCompletion --zsh bazel_src/scripts/zsh_completion/_bazel
```

Should do the trick with `installShellFiles` in `nativeBuildInputs`.

Comment (Profpatsch) [reply pkgs/development/tools/build-managers/bazel/bazel_3/default.nix:518]: Great! I always forget that this exists.

**Comment by Profpatsch** [comment]
Darwin failure is unrelated.

**Pull request merged**